internal/resources/kubernetessecret/resource_secret.go

```diff
@@ -34,52 +34,67 @@ func ResourceSecret() *schema.Resource {
 		DeleteContext: resourceSecretDelete,
 		UpdateContext: resourceSecretInPlaceUpdate,
 		ReadContext:   dataSourceSecretRead,
-		Schema:        secretSchema,
+		Schema:        getSecretSchema(false),
 	}
 }
 
-var secretSchema = map[string]*schema.Schema{
-	NameKey: {
-		Type:        schema.TypeString,
-		Description: "Name of the secret resource.",
-		Required:    true,
-		ForceNew:    true,
-		ValidateFunc: validation.All(
-			validation.StringLenBetween(1, 126),
-			validation.StringIsNotEmpty,
-			validation.StringIsNotWhiteSpace,
-		),
-	},
-	NamespaceNameKey: {
-		Type:        schema.TypeString,
-		Description: "Name of Namespace where secret will be created.",
-		Required:    true,
-		ForceNew:    true,
-		ValidateFunc: validation.All(
-			validation.StringIsNotEmpty,
-			validation.StringIsNotWhiteSpace,
-		),
-	},
-	OrgIDKey: {
-		Type:        schema.TypeString,
-		Description: "ID of Organization.",
-		Optional:    true,
-	},
-	scope.ScopeKey: scope.ScopeSchema,
-	specKey:        secretSpec,
-	common.MetaKey: common.Meta,
-	ExportKey: {
-		Type:        schema.TypeBool,
-		Description: "Export the secret to all namespaces.",
-		Optional:    true,
-		Default:     false,
-	},
-	statusKey: {
-		Type:        schema.TypeMap,
-		Description: "Status for the Secret Export.",
-		Computed:    true,
-		Elem:        &schema.Schema{Type: schema.TypeString},
-	},
+func getSecretSchema(isDataSource bool) map[string]*schema.Schema {
```

Comment:
This function is not intuitive. 
So, can you add functional pararement 
getSecretSchema(forDatasource), getSecretSchema(forResource)
Implement something in these lines 

or two new functions which will call this function

Reply:
Added two new function **getDataSourceSchema()** , **getResourceSchema()** which are calling **getSecretSchema(isDataSource bool)**

Reply:
PTAL